### PR TITLE
Feature:(issue_1927) Add ability for before handler to return new context

### LINKED ISSUE #1927: Ability to add context values
## Checklist

* [x] Are you running the latest v3 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v3 manual is [here](https://github.com/urfave/cli/blob/main/docs/v3/manual.md).
* [x] Did you perform a search about this feature? Here's the [GitHub guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

In old versions we could add context fields in Before function like this
```
  Before: func(c *cli.Context) error {
      db := "example"
      c.Context = context.WithValue(c.Context, "db", db)
      return nil
  },
```
It's useful for example for passing logger through all commands.

## Solution description
Add some middleware layer to command with signature
```
func(ctx context.Context, command *cli.Command) func(ctx context.Context, command *cli.Command)
```
where we could modify passing ctx

## Describe alternatives you've considered

Return back Context field to command

### PR DESCRIPTION
**Pull request by dearchap**
## What type of PR is this?

_(REQUIRED)_

- feature

## What this PR does / why we need it:

_(REQUIRED)_

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1927 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

go test -run=TestCommand_Run_BeforeReturnNewContext

## Release Notes

_(REQUIRED)_

```release-note
The Before handler for a command can optionally return a new context.Context which will then be propagated to the Action and After handlers. 
```

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -562,9 +562,11 @@ func (cmd *Command) Run(ctx context.Context, osArgs []string) (deferErr error) {
 	}
 
 	if cmd.Before != nil && !cmd.Root().shellCompletion {
-		if err := cmd.Before(ctx, cmd); err != nil {
+		if bctx, err := cmd.Before(ctx, cmd); err != nil {
 			deferErr = cmd.handleExitCoder(ctx, err)
 			return deferErr
+		} else if bctx != nil {
+			ctx = bctx
```
Comment (meatballhat) [inline command.go:569]: Is it not a legitimate usage for one's `Before` to return a new context *and* an error?

**Review by meatballhat — APPROVED** [review]
I like it! I had the one question and it looks like the docs tests failures are related to the change in func signature.

File: command.go (hunk 1 above)
Comment (dearchap) [reply command.go:569]: what would be the purpose of that since the app is going to return an error immediately anyway .

Comment (dearchap) [reply command.go:569]: Lets implement it if and when someone asks for it.

**Pull request merged**